pandas/core/generic.py

```diff
@@ -6354,11 +6169,7 @@ def _protect_consolidate(self, f):
         Consolidate _mgr -- if the blocks have changed, then clear the
         cache
         """
-        blocks_before = len(self._mgr.blocks)
-        result = f()
-        if len(self._mgr.blocks) != blocks_before:
-            self._clear_item_cache()
-        return result
+        return f()
```

Comment:
I think we don't need `_protect_consolidate` anymore right?

Reply:
good point, no